src/bin/common.rs

```diff
@@ -91,109 +108,59 @@ pub fn db_conn_string(config: &Config) -> Result<String, Error> {
 }
 
 #[derive(Debug)]
-pub struct GetError(String);
+pub struct MissingKeys(Vec<String>);
 
-impl fmt::Display for GetError {
+impl fmt::Display for MissingKeys {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "Could not find required key: {}", self.0)
+        write!(f, "Could not find required keys: [{}]", self.0.join(", "))
     }
 }
 
-impl StdError for GetError {
+impl StdError for MissingKeys {
```

Comment:
Any reason you're still keeping the `::std::error::Error` impls around?

Reply:
I'm trying to leverage Failure's Error/ErrorKind pair, and in order to maintain a context for the specific error, I.E. which fields are missing, I have a dedicated MissingKeys error which contains a Vec<String>

Reply:
I could implement Fail for that error instead

Reply:
fixed